design/authenticators_health_api.md

```diff
@@ -0,0 +1,233 @@
+person in another time. With the Healthcheck API the person who does the configuration will 
+get an immediate feedback on it, and the users that are *using* the authenticators will not 
+be unauthorized due to configuration errors.
+
+### Current Mechanism
+
+The current mechanism for understanding that there is a problem in the Authenticator configuration is as follows:
+
+1. The Operator configures the Authenticator according to the docs
+1. A user tries to authenticate\
+    1. this may be the Operator himself imitating a user to verify the authenticator works in the dev env
+1. The user gets a 401 Unauthorized response
+    1. There are other response codes possible but configuration issues will be 401
+1. The user reports to the operator that there is an issue
+    1. This one is tricky as the user might not know who to report this to.
+    1. Again, this may be the operator himself
+1. The operator looks in the logs (according to the docs) and finds the issue
+1. The operator fixes the issue
+
+This can be iterative as there might be several issues. It is possible to assume that in the case 
+above, the operator will solve this with the user and will ask him to try to authenticate again and 
+if it's still Unauthorized will solve the configuration issues one-by-one. 
+
+### Process Logic
+
+1. The Conjur operator logs into Conjur
+1. The Conjur operator configures the Authenticators according to the docs
+    1. We might have several authenticators configured
+1. The operator runs the Healthcheck request: `GET /authenticators/health`
+1. The operator gets a response:
+    1. In case all the authenticators are healthy, a response with code 200 and the following body:
+        
+        ```
+        {
+           “authenticators”:
+           {
+              "authn-1": "ok",
+              "authn-2": "ok"
+           },
+           "ok": true
+        }
+        ```
+    1. In case some of the authenticators aren't healthy, a response with code 500 (more info [here](authenticators_health_api.md#response-code-for-unhealthy-authenticators)) and the following body:
+
+        ```
+        {
+           “authenticators”:
+           {
+              "authn-1": "ok",
+              "authn-2": "<configuration error>",
+           },
+           "ok": true
+        }
+        ```
+        1. In case there is more than one error in an authenticator, they should be comma-separated.
+1. The operator understands the issue from the response and fixes the issue
+1. The operator runs the request once again and until he gets a healthy response
+
+**Note**: we should list in the response all the authenticators which are implemented in Conjur. 
+Authenticators for which we didn't implement the health-check will have the message "not-implemented". 
+A possible response can look like this:
+
+```
+{
+   “authenticators”:
+   {
+      "authn": "ok",
+      "authn-1": "ok",
+      "authn-2": "not-implemented"
+      "authn-3": "not-implemented",
```

Comment:
it would be interested to see here what it will look like given the revised response format above as well

Reply:
changed in the doc